File: codeforces/contest_round_579_div3/B/equal_rectangles2.cpp
#include <bits/stdc++.h>

using namespace std;

#define pb push_back
#define IOS ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0);
#define all(x) x.begin(),x.end()
#define debug(x) cout << x << endl;
#define FOR(x) for(int i =0; i < (x) ; i++)

int main(){
    IOS
    int T;
    cin >> T;
    for(int i = 0; i < T; i++){
        int n;
        cin >> n;
        //vector<int> arr;
        int num;
        long long total;
        for(int i = 0; i < 4*n; i++){
            cin >> num;
            total *= num;
        }
        if(pow(total, 1/float(n*2)) != int(pow(total, 1/float(n*2)))){
            cout << "NO" << endl;
        }
        else{
            cout << "YES" << endl;
        }
    }
    return 0;
}

File: codeforces/contest_round_579_div3/B/equal_rectangles2.py
def check_even(m):
    for value in m.values():
        if value % 2 != 0:
            return False
    return True
T = int(raw_input(""))
for i in range(T):
    n = int(raw_input(""))
    arr = map(int, raw_input("").split(" "))
    total = 1
    #print arr:
    m = {}
    for i in range(0, len(arr)):
        total *= arr[i]
        if m.get(arr[i], None) == None:
            m.update({arr[i] : 1})
        else:
            m[arr[i]] += 1
    if (total ** (1/float(n))) != int(total ** (1/float(n))) or check_even(m) == False:
        print "NO"
    else:
        print "YES"
